Tests/ApolloTests/RequestChainTests.swift

```diff
@@ -1215,4 +1215,193 @@ class RequestChainTests: XCTestCase {
     // then
     wait(for: expectations, timeout: 1, enforceOrder: true)
   }
+
+  // MARK: Response Tests
+
+  func test__response__givenUnsuccessfulStatusCode_shouldFail() throws {
+    // given
+    let client = MockURLSessionClient(
+      response: .mock(
+        url: TestURL.mockServer.url,
+        statusCode: 500,
+        httpVersion: nil,
+        headerFields: nil
+      ),
+      data: """
+      {
+        "data": {
+          "__typename": "Hero",
+          "name": "R2-D2"
+        }
+      }
+      """.data(using: .utf8)
+    )
+
+    let provider = DefaultInterceptorProvider(
+      client: client,
+      store: ApolloStore()
+    )
+
+    let transport = RequestChainNetworkTransport(
+      interceptorProvider: provider,
+      endpointURL: TestURL.mockServer.url
+    )
+
+    let expectation = expectation(description: "Response received")
+
+    _ = transport.send(operation: MockQuery<Hero>()) { result in
+      switch result {
+      case .success:
+        XCTFail("Unexpected response: \(result)")
+
+      case .failure:
+        expectation.fulfill()
+      }
+    }
+
+    wait(for: [expectation], timeout: 1)
+  }
+
+  // This test is odd because you might assume it would fail but there is no content-type checking on standard
+  // GraphQL response parsing. So this test is here to ensure that existing behaviour does not change.
+  func test__response__givenUnknownContentType_shouldNotFail() throws {
```

Comment:
I've added tests to keep the existing behaviour of standard GraphQL response parsing including this odd one.